x-pack/plugins/cases/server/services/index.ts

```diff
@@ -17,3 +18,30 @@ export { AttachmentService } from './attachments';
 export interface ClientArgs {
   unsecuredSavedObjectsClient: SavedObjectsClientContract;
 }
+
+export type ESConnectorFields = Array<{
+  key: string;
+  value: unknown;
+}>;
+
+export interface ESCaseConnector {
+  name: string;
+  type: ConnectorTypes;
+  fields: ESConnectorFields | null;
+}
+
+export interface CaseSavedObjectReference {
+  ref?: SavedObjectReference;
+  name: string;
+}
+
+/**
+ * The name of the saved object reference indicating the action connector ID. This is stored in the Saved Object reference
+ * field's name property.
+ */
+export const connectorIdReferenceName = 'connectorId';
```

Comment:
nit: `connectorIdReferenceName` -> `CONNECTOR_ID_REFERENCE_NAME`

Maybe we could move these variables to a `constants.ts` file in `x-pack/plugins/cases/server`